lib/class-wp-theme-json-gutenberg.php

```diff
@@ -207,57 +207,70 @@ class WP_Theme_JSON_Gutenberg {
 	 * - 'value': path to the value in theme.json and block attributes.
 	 */
 	const PROPERTIES_METADATA = array(
-		'background'       => array(
+		'background'           => array(
 			'value' => array( 'color', 'gradient' ),
 		),
-		'background-color' => array(
+		'background-color'     => array(
 			'value' => array( 'color', 'background' ),
 		),
-		'border-radius'    => array(
+		'border-radius'        => array(
 			'value' => array( 'border', 'radius' ),
 		),
-		'border-color'     => array(
+		// This is deliberately separate to the border-radius entry above.
+		// It allows for backwards compatible shorthand e.g. border-radius: 10px
+		// while below provides a means to explicitly set the CSS property name
+		// where the flat array configuration approach builds an incorrect name.
+		'corner-border-radius' => array(
+			'value'      => array( 'border', 'radius' ),
+			'properties' => array(
+				'border-top-left-radius'     => 'topLeft',
+				'border-top-right-radius'    => 'topRight',
+				'border-bottom-left-radius'  => 'bottomLeft',
+				'border-bottom-right-radius' => 'bottomRight',
+			),
+		),
```

Comment:
quick question here for @nosolosw @jorgefilipecosta what happens if you run this on top of `wp/trunk` I mean will the class from Gutenberg be used or the one from Core? I guess it should be Gutenberg's with the updates but just wanted to confirm.

Reply:
The plugin's code overwrites core's. In the front-end by hooking into the existing `wp_styles` ([code](https://github.com/WordPress/gutenberg/blob/trunk/lib/global-styles.php#L65)); in the editor by using the `__unstableType=globalStyles` ([code](https://github.com/WordPress/gutenberg/blob/trunk/lib/global-styles.php#L137)).